File: src/models/Vehicle.ts
export interface Vehicle {
    firstRegisteredAt: string;
    firstRegisteredInJerseyAt: string;
    make: string;
    model: string | null;
    type: VehicleType;
    color: string;
    cylinderCapacity: number | null;
    weight: string;
    co2Emissions: number | null;
    fuelType: VehicleFuelType;
    hash: string;
}

export interface VehicleWithLookupData extends Vehicle {
    previousOwners: number;
    previousTraders: number;
}

export enum VehicleFuelType {
    PETROL = "PETROL",
    HEAVY_OIL = "HEAVY_OIL",
    ELECTRIC = "ELECTRIC",
    HYBRID_ELECTRIC = "HYBRID_ELECTRIC",
    GAS = "GAS",
    DIESEL_ELECTRIC = "DIESEL_ELECTRIC",
    GAS_BI_FUEL = "GAS_BI_FUEL",
    STEAM = "STEAM",
    UNKNOWN = "UNKNOWN"
}

export enum VehicleType {
    CAR = "CAR",
    MOTORCYCLE = "MOTORCYCLE",
    AGRICULTURAL = "AGRICULTURAL",
    LIGHT_4X4_UTILITY = "LIGHT_4X4_UTILITY",
    LIGHT_COMMERCIAL = "LIGHT_COMMERCIAL",
    MEDIUM_WEIGHT = "MEDIUM_WEIGHT",
    HEAVY_WEIGHT = "HEAVY_WEIGHT",
    MOTOR_CARAVAN = "MOTOR_CARAVAN",
    MOTOR_DRAWN_TRAILER = "MOTOR_DRAWN_TRAILER",
    BUS = "BUS",
    CONSTRUCTION = "CONSTRUCTION",
    WORKS_TRUCK = "WORKS_TRUCK",
    UNKNOWN = "UNKNOWN"
}

File: src/main/java/je/glitch/data/api/models/Vehicle.java
package je.glitch.data.api.models;

import je.glitch.data.api.models.enums.VehicleFuelType;
import je.glitch.data.api.models.enums.VehicleType;
import lombok.Data;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Timestamp;

@Data
public class Vehicle {
    private final int id;
    private final Timestamp firstRegisteredAt;
    private final Timestamp firstRegisteredInJerseyAt;
    private final String make;
    private final String model;
    private final VehicleType type;
    private final String color;
    private final int cylinderCapacity;
    private final String weight;
    private final int co2Emissions;
    private final VehicleFuelType fuelType;

    public static Vehicle of(ResultSet result) throws SQLException {
        return new Vehicle(
                result.getInt("id"),
                result.getTimestamp("firstRegisteredAt"),
                result.getTimestamp("firstRegisteredInJerseyAt"),
                result.getString("make"),
                result.getString("model"),
                VehicleType.valueOf(result.getString("type")),
                result.getString("color"),
                result.getInt("cylinderCapacity"),
                result.getString("weight"),
                result.getInt("co2Emissions"),
                VehicleFuelType.valueOf(result.getString("fuelType"))
        );
    }
}
